### ISSUE TITLE
Get Element(s) from Shadow root using property

### ISSUE THREAD

**Issue by sarvaje** [w3c/webdriver#1610]
According to the documentation:
https://w3c.github.io/webdriver/#find-element-from-shadow-root step 2
https://w3c.github.io/webdriver/#find-elements-from-shadow-root step 2

the value for using has to be one of the [Locator strategies table](https://w3c.github.io/webdriver/#dfn-table-of-location-strategies).
But for what I can see in https://dom.spec.whatwg.org/ there no method to get elements from a tag name in a shadow root.

Experimenting in Chromium and Firefox I can neither use `xpath` in a shadow root. For the `xpath` I'm not sure if is because is not defined in the standard or because they don't have it implemented, but right now is not supported.

For more context you can take a look to [this thread](https://github.com/web-platform-tests/wpt/pull/29924#discussion_r686803204) in the WPT repository

**Comment by jimevans** [comment]
IIRC, there are some cases where using XPath outside of a shadow root will pierce into a shadow root to access elements (that's IIRC, and I may not be "recalling correctly"). This seems to indicate that it's _possible_ to use XPath for a shadow root, at least in some cases.

Not having these two selector strategies makes the "Find Element(s) From Shadow Root" inconsistent with the other "Find Element(s)" commands, and that inconsistency should be avoided. Just because something can't be done via JavaScript doesn't mean it should be disallowed via WebDriver. It may not be _easy_ for an implementer, but users shouldn't be penalized for that difficulty. I guarantee that supporting some but not all locator strategies will be a source of issue reports for implementers, even if we modify the spec to disallow the more challenging ones.

**Comment by AutomatedTester** [comment]
Unless I am misunderstanding, we should have support tag name as it is part of the [`DocumentOrShadowRoot`](https://dom.spec.whatwg.org/#ref-for-documentorshadowroot) mixin. If it's not working then we need to seek clarifications from DOM.

**Comment by sarvaje** [comment]
> Not having these two selector strategies makes the "Find Element(s) From Shadow Root" inconsistent with the other "Find Element(s)"...

I don't agree 100%, IMO it would be inconsistent if they were the same, but an Element and a shadow root are not the same thing.
Also, even if the name is similar (“Find Elements”), the method you are using is not the same. Developers will be using different methods to get a shadow root or an element, so IMO, it is ok if those methods have different parameters.

**Comment by jimevans** [comment]
A shadow root may not be the same as an element, but it most certainly is analogous* to a document, from which one can also find an element. I stand by my analysis that not having this parity will be an unending source of bug reports from users who will not appreciate the inconsistency. This applies both to client implementations like Selenium, and to browser maintainers who will eventually get the upstream issue reports.

*”Analogous,” not identical, in case others are feeling pedantic.

**Comment by whimboo** [comment]
We should also discuss if drivers should return a `unsupported operation` error in case the search strategy is actually not supported.

**Comment by jimevans** [comment]
If a browser cannot support a location strategy, as disappointing as that is, I would support a remote end returning `unsupported operation` as a compromise.

**Comment by beliaev-maksim** [comment]
@whimboo yes
I spawned the discussion in Selenium and we agreed that it is better to discuss in webdriver community.
https://github.com/SeleniumHQ/selenium/pull/11893

**Comment by gsnedders** [comment]
Some of this problem is the fact that XPath's DOM integration, especially regards to shadow roots, is relatively undefined.

Like, `document.evaluate()` has an arg (its second) which is `Node contextNode`. On the face of it, therefore, you should be able to pass a `ShadowRoot`.

It might be worth filing a bug on the DOM spec if this doesn't work, or at least write some tentative tests that shows the current behaviour of browsers.
